webapp/src/Controller/API/ContestController.php

```diff
@@ -332,59 +332,106 @@ public function setBannerAction(Request $request, string $cid, ValidatorInterfac
+     *                 description="The new unfreeze (thaw) time of the contest",
+     *                 type="string",
+     *                 format="date-time"
      *             )
      *         )
      *     )
      * )
      * @OA\Response(
      *     response="200",
-     *     description="Contest start time changed successfully",
+     *     description="Contest object if it changed",
      *     @OA\JsonContent(
-     *         type="string"
+     *         allOf={
+     *             @OA\Schema(ref=@Model(type=Contest::class)),
+     *             @OA\Schema(ref="#/components/schemas/Banner")
+     *         }
      *     )
      * )
+     * @OA\Response(
+     *     response="204",
+     *     description="The change was successful"
+     * )
      */
-    public function changeStartTimeAction(Request $request, string $cid): Response
+    public function changeTimesAction(Request $request, string $cid): Response
     {
         $contest  = $this->getContestWithId($request, $cid);
-        $now      = Utils::now();
+        $now      = (int)Utils::now();
         $changed  = false;
         if (!$request->request->has('id')) {
             $response = new JsonResponse('Missing "id" in request.', Response::HTTP_BAD_REQUEST);
-        } elseif (!$request->request->has('start_time')) {
-            $response = new JsonResponse('Missing "start_time" in request.', Response::HTTP_BAD_REQUEST);
+        } elseif (!$request->request->has('start_time') && !$request->request->has('scoreboard_thaw_time')) {
+            $response = new JsonResponse('Missing "start_time" or "scoreboard_thaw_time" in request.', Response::HTTP_BAD_REQUEST);
         } elseif ($request->request->get('id') != $contest->getApiId($this->eventLogService)) {
             $response = new JsonResponse('Invalid "id" in request.', Response::HTTP_BAD_REQUEST);
-        } elseif (!$request->request->has('force') &&
-            $contest->getStarttime() != null &&
-            $contest->getStarttime() < $now + 30) {
-            $response = new JsonResponse('Current contest already started or about to start.',
-                                         Response::HTTP_FORBIDDEN);
-        } elseif ($request->request->get('start_time') === null) {
-            $this->em->persist($contest);
-            $contest->setStarttimeEnabled(false);
-            $response = new JsonResponse('Contest paused :-/.', Response::HTTP_OK);
-            $this->em->flush();
-            $changed = true;
-        } else {
-            $date = date_create($request->request->get('start_time'));
-            if ($date === false) {
-                $response = new JsonResponse('Invalid "start_time" in request.', Response::HTTP_BAD_REQUEST);
+        } elseif ($request->request->has('start_time')) {
+            // By default, it is not allowed to change the start time in the last 30 seconds before contest start.
+            // We allow the "force" parameter to override this.
```

Comment:
I'm not sure if I like such a parameter. For the (un)locking of contests we also decided against such a parameter.

Reply:
We already had it for start time change. We should be consistent and drop both or keep both